### DISCUSSION TITLE
Suggestion - Str::splitCamelCase()

### DISCUSSION THREAD

**Discussion by omaromp2** [laravel/framework#48744] [Ideas]
A method to split a camel Case string into individual words. This can be useful for various tasks like generating human-readable labels from variable or function names.

Example: 
``` php
 echo Str::splitCamelCase('camelCaseString'); // Outputs: camel Case String
```

**Comment by shaedrich** [comment]
Wouldn't that be simply something like `Str::spaceCase()` or `Str::gapCase()` since I would expect a method called "split" would return an array.

**Comment by juse-less** [comment]
Not exactly what you're after, but in case you've missed it; the [`Str::headline()`](https://laravel.com/docs/10.x/strings#method-str-headline) helper does split text based on casing, hyphen, and underscore.
The caveat being that it upper-case every word.

**Comment by omaromp2** [comment]
Thank you for your recommendations. I'm going to test the headline method to see if it can achieve what I wanted to do. I'm just looking for ways to contribute to the framework. 👍

**Comment by omaromp2** [comment]
Yup, headline basically does the same.
